qualtran/bloqs/util_bloqs.py

```diff
@@ -138,6 +139,38 @@ def wire_symbol(self, soq: 'Soquet') -> 'WireSymbol':
         return directional_text_box(' ', side=soq.reg.side)
 
 
+@frozen
+class Partition(Bloq):
+    """Partition a generic index into multiple registers.
+
+    Args:
+        n: The total bitsize of the un-partitioned register
+        regs: Registers to partition into. The `side` attribute is ignored.
+        partition: `False` means un-partition instead.
+
+    Registers:
+        x: the un-partitioned register. LEFT by default.
+        [user spec]: The registers provided by the `regs` argument. RIGHT by default.
+    """
+
+    n: int
+    regs: Tuple[Register, ...]
+    partition: bool = True
+
+    @cached_property
+    def signature(self) -> 'Signature':
```

Comment:
Can we implement `as_cirq_op` so that it does the correct reshaping of cirq quregs, similar to `Split` and `Join` ?